src/handlers/chain_agnostic/mod.rs

```diff
@@ -145,3 +168,102 @@ pub async fn check_bridging_for_erc20_transfer(
     }
     Ok(None)
 }
+
+/// Compute the simulation state override balance for a given balance
+pub fn compute_simulation_balance(balance: u128) -> Bytes {
+    let raw_bytes = balance.to_be_bytes();
+    let mut padded = [0u8; 32];
+    padded[32 - raw_bytes.len()..].copy_from_slice(&raw_bytes);
+    Bytes::from(padded.to_vec())
+}
+
+/// Compute the storage slot for a given address and slot number to use in the
+/// simulation state overrides
+/// https://docs.tenderly.co/simulations/state-overrides#storage-slot-calculation
+pub fn compute_simulation_storage_slot(address: Address, slot_number: u64) -> Bytes {
+    let address_string = address.to_string();
+    let address_clean = address_string.trim_start_matches("0x").to_lowercase();
+    let padded_address = format!("{:0>64}", address_clean);
+    let slot_hex = format!("{:x}", slot_number);
+    let padded_slot = format!("{:0>64}", slot_hex);
+    let concatenated = format!("{}{}", padded_address, padded_slot);
+    let concatenated_bytes = hex::decode(concatenated).expect("Failed to decode hex");
+    let result = keccak256(concatenated_bytes);
+    Bytes::from(result.to_vec())
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Erc20AssetChange {
+    pub chain_id: String,
+    pub asset_contract: Address,
+    pub amount: U256,
+}
+
+/// Get the ERC20 assets changes from the transaction simulation result
+pub async fn get_assets_changes_from_simulation(
+    simulation_provider: Arc<dyn SimulationProvider>,
+    transaction: &Transaction,
+) -> Result<Vec<Erc20AssetChange>, RpcError> {
+    // Fill the state overrides for the source address for each of the supported
+    // assets on the initial tx chain
+    let state_overrides = {
+        let mut state_overrides = HashMap::new();
+        let assets_contracts =
+            get_bridging_assets_contracts_for_chain(&transaction.chain_id.clone());
+        let mut account_state = HashMap::new();
+        account_state.insert(
+            // Since we are using only USDC for the bridging,
+            // we can hardcode the storage slot for the contract which is 9
+            compute_simulation_storage_slot(transaction.from, 9),
+            compute_simulation_balance(99000000000),
+        );
+        for contract in assets_contracts {
+            state_overrides.insert(
+                Address::from_str(&contract).unwrap_or_default(),
```

Comment:
I think we should to be offensive here, not defensive? Anyway, why isn't `contract` already an `Address` which would avoid this unwrap?

```suggestion
                Address::from_str(&contract).unwrap(),
```

Reply:
Good catch, thank you! Updated to use `Address` everywhere instead.